book/accessing_argv.md

```diff
@@ -0,0 +1,98 @@
+# Accessing ARGV
+
+If you have positional and REST parameters along with a flag, you can access them in your main script file in the `main` function - unlike other scripting languages, Nushell doesn't provide direct access to the ARGV input array, but here is a way to circumvent the issue:
+
+```nu
+# supercar.nu
+
+def main [
+  manuf: string
+  model: string
+  --fast (-f)
+  ] {
+    [$manuf $model $fast] | flatten
+}
+```
+
+```nu
+nu supercar.nu manuf=ferrari model=F40 -f
+
+╭───┬───────────────╮
+│ 0 │ manuf=ferrari │
+│ 1 │ model=F40     │
+│ 2 │ true          │
+╰───┴───────────────╯
+
+nu supercar.nu manuf=citroen model=2cv
+
+╭───┬──────────────────╮
+│ 0 │ manuf=citroen    │
+│ 1 │ model=2cv        │
+│ 2 │ false            │
+╰───┴──────────────────╯
+```
+
+Once your command line becomes complicated, the parameter list very quickly becomes unwieldy:
+
+```nu
+def main [
+  manuf: string
+  model: string
+  --fast (-f)
+  --num-cylinders: int,
+  --color: string
+  ] {
+    [$manuf $model $fast $num_cylinders $color] | flatten
+}
+
+nu supercar.nu manuf=ferrari model=F40 -f --color=red
```

Comment:
i'm not sure why you would call things like this :thinking: 

are we talking about keyword arguments?

Reply:
It's just a way of keeping the key and value together! Putting a space between them leads to this:
╭───┬─────────╮
│ 0 │ manuf   │
│ 1 │ ferrari │
│ 2 │ true    │
│ 3 │ model   │
│ 4 │ F40     │
│ 5 │ --color │
│ 6 │ red     │
╰───┴─────────╯
Which is, in my opinion as a total beginner with Nushell, even worse for associating the keyword with the value. (I still haven't figured out how to separate "manuf=ferrari" into a record! For my own script, I feed the result into a Typescript script, and that does all the separating and parsing for me.) Besides which, I've never actually been a fan of the "key-space-value" format, although I do know some (many?) folk prefer it.

Now to figure out how to upload the edited files... 😛